### DISCUSSION TITLE
Azure Push Hub - Does Shiny support Push notification Templates ?

### DISCUSSION THREAD

**Discussion by MGohil** [shinyorg/shiny#1289] [Q&A] [answered]
Microsoft Azure Push hub supports Templates for push notification as per below blog post:
https://devblogs.microsoft.com/azure-notification-hubs/use-templates-effectively/

Does Shiny supports setting up these templates as per below?
![image](https://github.com/shinyorg/shiny/assets/16605284/7b88d629-3169-4f44-8270-937930a0e257)

This requires to get an existing instance of NotificationHub in the client app, but Is there any way I can get this instance and then do this Template registration per platform in my app?

If Shiny doesn't expose the existing instance of Hub, can we still have new instance of Hub, using same ConnectionString and Hubname in app and do this template registration without any issues?

Thanks,
Milan G

**Comment by aritchie** [comment]
Not out of the box.  v3 of Shiny does have something called a push provider which is meant to be a small implementations on top of the native registration bits.  You can easily plugin additional requirements or registration constraints there and still let Shiny manage everything underneath.

**Reply by aritchie** [reply, marked as answer]
This is now documented how you can have a custom push provider for more custom scenarios: https://shinylib.net/client/push/providers/

**Reply by MGohil** [reply]
Thank you @aritchie for quick turnaround. I was just wondering how to achieve this but Thanks for the sample in documentation.
